components/bitcoin-topics/Navigation.tsx

```diff
@@ -40,7 +47,23 @@ export function Navigation({
     onLinkClick?: React.MouseEventHandler<HTMLAnchorElement>
 }) {
     const pathname = usePathname()
-    const posts = allCoreContent(sortPosts(allTopics))
+    const posts = useMemo(() => allCoreContent(sortPosts(allTopics)), [])
+    const [expandedTopics, setExpandedTopics] = useState<
+        Record<string, boolean>
+    >({})
+
+    // Load expanded state from localStorage on component mount
+    useEffect(() => {
+        const savedState = localStorage.getItem("expandedTopics")
+        if (savedState) {
+            setExpandedTopics(JSON.parse(savedState))
```

Comment:
parsing JSON should be in a try/catch block. There is always a chance to error out if it parses incorrect json formatted string. Given its pulling its data from local storage. The string cannot be trusted to always be properly formatted.